packages/tupaia-web/src/features/Map/MapOverlaySelector/DesktopMapOverlaySelector.tsx

```diff
@@ -146,47 +164,61 @@ export const DesktopMapOverlaySelector = ({
 }: DesktopMapOverlaySelectorProps) => {
   const { projectCode, entityCode } = useParams();
   const { hasMapOverlays } = useMapOverlays(projectCode, entityCode);
-  const { period } = useActiveMapOverlayReport();
+  const { measureData, period } = useMapOverlayData();
+  const [mapModalOpen, setMapModalOpen] = useState(false);
+  const toggleMapTableModal = () => {
+    setMapModalOpen(!mapModalOpen);
+  };
 
   return (
-    <Wrapper>
-      <Header>
-        <Heading>Map Overlays</Heading>
-      </Header>
-      <Container>
-        <TitleWrapper>
-          <MapOverlaySelectorTitle />
-          <MapOverlayDatePicker />
-        </TitleWrapper>
-        {hasMapOverlays && (
-          <OverlayLibraryAccordion
-            expanded={overlayLibraryOpen}
-            onChange={toggleOverlayLibrary}
-            square
-          >
-            <OverlayLibraryHeader
-              expandIcon={<ArrowDropDown />}
-              aria-controls="overlay-library-content"
-              id="overlay-library-header"
+    <>
+      {mapModalOpen && <MapTableModal onClose={toggleMapTableModal} />}
+      <Wrapper>
+        <Header>
+          <Heading>Map Overlays</Heading>
+          {measureData && (
+            <MapTableButton>
+              <Tooltip arrow interactive placement="top" title="Generate Report">
+                <TableAssignmentIcon onClick={toggleMapTableModal} />
```

Comment:
the `onClick` should be on the `MapTableButton` instead of the `TableAssignmentIcon`